File: src/main/java/epicsquid/superiorshields/enchantment/QuickenedEnchantment.java
package epicsquid.superiorshields.enchantment;

import net.minecraft.world.item.enchantment.EnchantmentCategory;

public class QuickenedEnchantment extends ShieldEnchantment {

	public QuickenedEnchantment(Rarity rarityIn, EnchantmentCategory typeIn) {
		super(rarityIn, typeIn);
	}

	@Override
	public int getMaxLevel() {
		return 3;
	}
}

File: src/main/kotlin/dev/epicsquid/superiorshields/enchantment/QuickenedEnchantment.kt
package dev.epicsquid.superiorshields.enchantment

import dev.epicsquid.superiorshields.config.Config
import dev.epicsquid.superiorshields.effects.EffectHandler
import net.minecraft.world.item.enchantment.EnchantmentCategory
import kotlin.math.roundToInt

class QuickenedEnchantment(
	rarity: Rarity,
	category: EnchantmentCategory,
	effectHandler: EffectHandler
) : SuperiorShieldEnchantment(rarity, category, effectHandler), AttributeProvider {
	override fun getMaxLevel(): Int = 3

	override fun shieldAttributeModifiers(level: Int): ShieldAttributeModifiers =
		ShieldAttributeModifiers(
			rechargeRateMultiplier = level.toDouble() * Config.SHIELDS_CONFIG.quickenMultiplier.get()
		)
}